### ISSUE TITLE
Playing Sample Track inside BB Track does not always work

### ISSUE THREAD

**Issue by karmux** [LMMS/lmms#4422]
1. If BB Track is played from the middle then Sample Track inside BB Track is silent.

2. If BB Track is played in the loop and loop is not in the beginning of the song then for first time Sample Track inside BB Track is silent and starts playing only when
a) playhead moves back to beginning of loop or
b) playhead starts playing sample again in a longer loop and pattern

Example project: [st-in-bbt.zip](https://github.com/LMMS/lmms/files/2091593/st-in-bbt.zip) (1.1 MB)

**Comment by PhysSong** [comment]
Tagging @BaraMGB, the author of #3133.

**Comment by JohannesLorenz** [comment]
I'm not sure if this is a bug, and to me it's at least not critical enough for a stable-1.2 bug (and the PR is quite large/dangerous for a stable-1.2 fix).

I think it might not be a bug because, if you play the piano roll and start in the middle of a note, that note is not being played. So equivalently, if you play the BB editor and start in the middle of a sample, that sample should maybe not be played?

Opinions?

**Comment by BaraMGB** [comment]
The pull request I had delivered is far from being stable. In my opinion this issue is a feature request.

**Comment by JohannesLorenz** [comment]
@karmux Since you opened the issue: Can you please also check if you agree/disagree with my previous post?

**Comment by karmux** [comment]
@JohannesLorenz I agree that making LMMS to play Sample Track inside BB Track turned out to require bigger changes in the code which is more like a new feature than bugfix.

**Comment by JohannesLorenz** [comment]
Many thanks both of you. I changed the tag to "enhancement" and will ask BaraMGB to retarget the PR to master.
